packages/main/src/plugin/extensions-updater/extensions-updater.ts

```diff
@@ -0,0 +1,230 @@
+          default: true,
+        },
+      },
+    };
+
+    this.configurationRegistry.registerConfigurations([extensionLoaderConfiguration]);
+
+    // check on configuration change
+    this.configurationRegistry.onDidChangeConfiguration(async event => {
+      if (event.key === autoCheckUpdatesKey && event.value === true) {
+        await this.checkForUpdates();
+      }
+
+      if (event.key === autoUpdateKey && event.value === true) {
+        await this.checkForUpdates();
+      }
+    });
+
+    // setup recurring check
+    this.intervalChecker = setInterval(() => {
+      this.checkForUpdates().catch((err: unknown) => {
+        console.error('Error while checking for updates', err);
+      });
+    }, ExtensionsUpdater.CHECK_FOR_UPDATES_INTERVAL);
+
+    // check on startup
+    await this.checkForUpdates();
+  }
+
+  async stop() {
+    if (this.intervalChecker) {
+      clearInterval(this.intervalChecker);
+    }
+  }
+
+  isAutoCheckUpdatesEnabled(): boolean {
+    const config = this.configurationRegistry.getConfiguration(ExtensionsUpdaterSettings.SectionName);
+    return config.get(ExtensionsUpdaterSettings.AutoCheckUpdates) === true;
+  }
+
+  isAutoUpdateEnabled(): boolean {
+    const config = this.configurationRegistry.getConfiguration(ExtensionsUpdaterSettings.SectionName);
+    return config.get(ExtensionsUpdaterSettings.AutoUpdate) === true;
+  }
+
+  async checkForUpdates(): Promise<void> {
+    if (this.isAutoCheckUpdatesEnabled() || this.isAutoUpdateEnabled()) {
+      try {
+        await this.doCheckForUpdates();
+      } catch (err) {
+        console.error('Error while checking for updates', err);
+      }
+    }
+  }
+
+  // check if some extensions can be updated or not
+  async doCheckForUpdates(): Promise<void> {
+    // grab list of extensions
+    const availableExtensions = await this.extensionCatalog.getExtensions();
+
+    // now, grab list of installed extensions
+    const installedExtensions = await this.extensionLoader.listExtensions();
+
+    // now, for each installed extension that is not a built-in extension, check if there is a newer version available
+    const extensionsToUpdate = installedExtensions
+      .filter(extension => extension.removable === true)
+      .map(installedExtension => {
+        // find the extension in the list of available extensions
+        const availableExtension = availableExtensions.find(extension => extension.id === installedExtension.id);
+        // not found? skip
+        if (!availableExtension) {
+          console.log(
+            `Skipping update for extension ${installedExtension.id} because it is not available in the registry`,
+          );
+          return undefined;
+        }
+
+        // if found compare versions
+        const installedVersion = installedExtension.version;
+        // take latest version
+        const latestAvailableVersion = availableExtension.versions.filter(version => version.preview === false)?.[0];
```

Comment:
During testing I've tried to downversion the RedHat authentication extension to 0.0.0 and it didn't show the update button. By digging i discovered that it misses the preview tag from https://registry.podman-desktop.io/api/extensions.json. The question is, is it just a bug of the json file or it could be possible to omit the preview tag and consequently if it is omitted it's like true?

Reply:
> The question is, is it just a bug of the json file or it could be possible to omit the preview tag and consequently if it is omitted it's like true?

bug of the json file